sktime/forecasting/regularized_var.py

```diff
@@ -0,0 +1,224 @@
+
+
+class RegularizedVAR(BaseForecaster):
+    """Custom forecaster using VAR with regularization.
+
+    Parameters
+    ----------
+    lags : int
+        The number of lagged values to include in the model.
+    L1_penalty : float, optional (default=0.0)
+        The L1 regularization penalty.
+    L2_penalty : float, optional (default=0.0)
+        The L2 regularization penalty.
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["meraldoantonio"],
+        "python_dependencies": "cvxpy",
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+    }
+
+    def __init__(self, lags, L1_penalty=0.0, L2_penalty=0.0):
+        self.lags = lags
+        self.L1_penalty = L1_penalty
+        self.L2_penalty = L2_penalty
+        self.coefficients = None
+        self.intercept = None
+        self.num_series = None
+        super().__init__()
+
+    def create_var_data(self, data):
+        """
+        Prepare the data for VAR(p) model.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            The input time series data.
+
+        Returns
+        -------
+        X : np.ndarray
+            The lagged values as predictors.
+        y : np.ndarray
+            The current values as response variable.
+        """
+        df = pd.concat([data.shift(i) for i in range(self.lags + 1)], axis=1)
+        df.columns = [
+            f"{col}_lag{i}" for i in range(self.lags + 1) for col in data.columns
+        ]
+        df = df.dropna()
+        y = df[[f"{col}_lag0" for col in data.columns]].values
+        X = df.drop(columns=[f"{col}_lag0" for col in data.columns]).values
+        return X, y
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Guaranteed to have a single column if scitype:y=="univariate".
+        fh : ForecastingHorizon, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        import cvxpy as cp
+
+        X, y = self.create_var_data(y)
+        n, k = X.shape
+        num_series = y.shape[1]
+        self.num_series = num_series
+
+        # Define the optimization problem
+        coefficients = cp.Variable((k, num_series))
+        intercept = cp.Variable((1, num_series))
+
+        # Define the objective function with regularization
+        objective = cp.Minimize(
+            cp.sum_squares(y - X @ coefficients - intercept)
+            + self.L1_penalty * cp.norm(coefficients, 1)
+            + self.L2_penalty * cp.norm(coefficients, "fro")
+        )
+        problem = cp.Problem(objective)
+        problem.solve()
```

Comment:
this is the same as `sklearn` `ElasticNet`, from a math perspective, so questions (I have no clear opinion yet):

* use `sklearn` `ElasticNet` instead?
* should there be a "regressor" instead in the parameter, defaulting to `ElasticNet`, and the user can choose anything else `sklearn` compatible? This could be VARreduce or similar

Reply:
That makes a lot of sense! I've reworked the module; now instead of using `cvxpy`, users can specify a regressor instead. 
I chose `LinearRegression` as the default as this will make `VARReduce` identical to `VAR`, which might avoid confusion.

Reply:
yes, good choice of default, better than my suggestion!
And users can use `ElasticNet` if they also want L1 penalty.